### DISCUSSION TITLE
[Bug Report] Trajectory is not replayed correctly

### DISCUSSION THREAD

**Discussion by xzhu0428** [isaac-sim/IsaacLab#1122] [Q&A]
### Describe the bug

I'm using IsaacLab as the dynamics function in an MPC framework but encountered issues with trajectory replay.
In the Ant environment, I rolled out the trajectory for states and actions: $s_1, a_1, …, s_H, a_H$. When I reset the environment to state $s_{10}$ and execute the actions $a_{10}, …, a_H$, the resulting trajectory deviates from the expected states $s_{11}, …, s_H$. However, when starting from $s_1$ and replaying the entire trajectory, the results are as expected. This issue doesn't occur in the Cartpole environment.

Additional experiments:
- Running the same trajectory sequence in parallel environments results in different trajectories across all environments.
- Raising the Ant to 5m above ground and disabling gravity significantly reduces the replay error, but the discrepancy is still present.

### Steps to reproduce

Copy and paste code in `.txt` to `.py`: [test_ant.txt](https://github.com/user-attachments/files/17134718/test_ant.txt)

### System Info

Describe the characteristic of your environment:

- Commit: 59fd1f7f9d6a1ad55e519751e104e1f94ef60512
- Isaac Sim Version: 4.2.0-rc.18+release.16044.3b2ed111.gl
- OS: Ubuntu 22.04
- GPU: Nvidia L4
- CUDA: 12.5
- GPU Driver: 555.42.06

### Checklist

- [x] I have checked that there is no similar issue in the repo (**required**)
- [ ] I have checked that the issue is not in running Isaac Sim itself and is related to the repo

### Acceptance Criteria

Add the criteria for which this task is considered **done**. If not known at issue creation time, you can add this once the issue is assigned.

- [ ] Criteria 1
- [ ] Criteria 2

**Comment by kellyguo11** [comment]
Hello, to guarantee determinism, the simulation should be stopped and restarted for each trajectory. In addition, we are aware that spawning environments away from the origin, which is often the case when running with parallel environments, can cause floating point precision errors in simulation that eventually propagate to larger errors. In this case, it is recommended to spawn all environments at the world origin.

**Reply by xzhu0428** [reply]
Hi Kelly, thanks for the reply! I'll change the spawn location to the world origin. 
I tried to restart the simulation for the replay experiment (see below), but still cannot replay the trajectory.

> In the Ant environment, I rolled out the trajectory for states and actions: $s_1, a_1, ..., s_H, a_H$. When I reset the environment to state $s_{10}$ and execute the actions $a_10, ..., a_H$, the resulting trajectory deviates from the expected states $s_{11}, ..., s_H$.
